From b11aa8bc017c8696d990cff398441690b707fc91 Mon Sep 17 00:00:00 2001
From: Zhihao Cheng <chengzhihao1@huawei.com>
Date: Sat, 17 Jun 2023 18:57:32 +0800
Subject: [PATCH 037/780] [Huawei] Revert "[Huawei] ext4: Limit retry times if
 ext4_writepages occurs ENOSPC"

Offering: HULK
hulk inclusion
category: bugfix
bugzilla: 188819

--------------------------------

This reverts commit c9d2a3f9a0605ef66d2faeef38b75cad605a87a3.

Signed-off-by: Zhihao Cheng <chengzhihao1@huawei.com>
---
 fs/ext4/inode.c | 8 +++-----
 1 file changed, 3 insertions(+), 5 deletions(-)

diff --git a/fs/ext4/inode.c b/fs/ext4/inode.c
index 708cc36d2a37..388b13f37bd3 100644
--- a/fs/ext4/inode.c
+++ b/fs/ext4/inode.c
@@ -2450,7 +2450,7 @@ static int mpage_map_one_extent(handle_t *handle, struct mpage_da_data *mpd)
  */
 static int mpage_map_and_submit_extent(handle_t *handle,
 				       struct mpage_da_data *mpd,
-				       bool *give_up_on_write, int *retries)
+				       bool *give_up_on_write)
 {
 	struct inode *inode = mpd->inode;
 	struct ext4_map_blocks *map = &mpd->map;
@@ -2478,8 +2478,7 @@ static int mpage_map_and_submit_extent(handle_t *handle,
 			 * is non-zero, a commit should free up blocks.
 			 */
 			if ((err == -ENOMEM) ||
-			    (err == -ENOSPC && EXT4_SB(sb)->s_mb_free_pending &&
-			     ++(*retries) < 3)) {
+			    (err == -ENOSPC && EXT4_SB(sb)->s_mb_free_pending)) {
 				if (progress)
 					goto update_disksize;
 				return err;
@@ -2696,7 +2695,6 @@ static int ext4_writepages(struct address_space *mapping,
 	struct ext4_sb_info *sbi = EXT4_SB(mapping->host->i_sb);
 	struct blk_plug plug;
 	bool give_up_on_write = false;
-	int __maybe_unused retries = 0;
 
 	if (unlikely(ext4_forced_shutdown(EXT4_SB(inode->i_sb))))
 		return -EIO;
@@ -2842,7 +2840,7 @@ static int ext4_writepages(struct address_space *mapping,
 		ret = mpage_prepare_extent_to_map(&mpd);
 		if (!ret && mpd.map.m_len)
 			ret = mpage_map_and_submit_extent(handle, &mpd,
-					&give_up_on_write, &retries);
+					&give_up_on_write);
 		/*
 		 * Caution: If the handle is synchronous,
 		 * ext4_journal_stop() can wait for transaction commit
-- 
2.17.1

